executor_manager/executors/docker/executor.py

```
@@ -371,21 +392,54 @@ def _check_container_health(
+            )
+            # Clean up the container on timeout
+            try:
+                self.subprocess.run(
+                    ["docker", "rm", "-f", executor_name],
+                    capture_output=True,
+                    timeout=10,
+                )
+                logger.info(
+                    f"Cleaned up container after health check timeout: {executor_name}"
+                )
+            except Exception as cleanup_error:
```

Ruff (0.14.11)

Do not catch blind exception: `Exception`

(BLE001)